app/src/main/java/org/wikipedia/donate/donationreminder/DonationReminderViewModel.kt

```diff
@@ -0,0 +1,147 @@
+    }
+
+    fun updateReadFrequencyState(readFrequency: Float) {
+        _uiState.update { it.copy(readFrequency = it.readFrequency.copy(selectedValue = readFrequency)) }
+    }
+
+    fun toggleDonationReminders(enabled: Boolean) {
+        Prefs.donationReminderConfig = Prefs.donationReminderConfig.copy(isEnabled = enabled)
+        _uiState.update { it.copy(isDonationReminderEnabled = enabled) }
+    }
+
+    private fun createReadFrequencyOptions(): SelectableOption {
+        val options = DonationReminderHelper.defaultReadFrequencyOptions
+        val optionItems = options.map {
+            OptionItem.Preset(it, "${it.toInt()} articles")
+        } + OptionItem.Custom
+
+        val selectedValue = if (Prefs.donationReminderConfig.articleFrequency <= 0) options.first()
+        else Prefs.donationReminderConfig.articleFrequency
+
+        return SelectableOption(
+            selectedValue,
+            optionItems,
+            minimumAmount = 1f,
+            maximumAmount = 1000f
+        ) {
+            "${it.toInt()} articles"
+        }
+    }
+
+    private suspend fun createDonationAmountOptions(): SelectableOption {
+        val donationConfig = DonationConfigHelper.getConfig()
+        val minimumAmount = donationConfig?.currencyMinimumDonation?.get(currencyCode) ?: 0f
+
+        var maximumAmount = donationConfig?.currencyMaximumDonation?.get(currencyCode) ?: 0f
+        if (maximumAmount == 0f) {
+            val defaultMin = donationConfig?.currencyMinimumDonation?.get(GooglePayComponent.CURRENCY_FALLBACK) ?: 0f
+            if (defaultMin > 0f) {
+                maximumAmount = (donationConfig?.currencyMinimumDonation?.get(currencyCode) ?: 0f) / defaultMin *
+                        (donationConfig?.currencyMaximumDonation?.get(GooglePayComponent.CURRENCY_FALLBACK) ?: 0f)
+            }
+        }
+
+        val presets = DonationReminderHelper.currencyAmountPresets[currentCountryCode] ?: listOf(0f)
+        val options = presets.map {
+            OptionItem.Preset(it, currencyFormat.format(it).replace(Regex("\\.00$"), ""))
+        } + OptionItem.Custom
+
+        val selectedValue = if (Prefs.donationReminderConfig.donateAmount <= 0f) presets.first()
+        else Prefs.donationReminderConfig.donateAmount
+
+        return SelectableOption(
+            selectedValue,
+            options,
+            minimumAmount = minimumAmount,
+            maximumAmount = maximumAmount,
+            displayFormatter = {
+                currencyFormat.format(it).replace(Regex("\\.00$"), "")
+            }
+        )
+    }
+
+    fun getAmountFloat(text: String): Float {
+        var result: Float?
+        result = text.toFloatOrNull()
+        if (result == null) {
+            val text2 = if (text.contains(".")) text.replace(".", ",") else text.replace(",", ".")
+            result = text2.toFloatOrNull()
+        }
+        return result ?: 0f
+    }
```

Comment:
This is a duplicate function from `GooglePayAcitivty`. Perhaps we can consolidate this and the currency-related variable into a `DonateUtil` class, allowing us to share the same functionalities in the future.